### DISCUSSION TITLE
Supabase "createClient" sending too many requests in network tab

### DISCUSSION THREAD

**Discussion by GreatHayat** [supabase/supabase#18390] [Questions]
Hello everyone!

I just started with supabase in `nextjs 13` (App Directory) and I used `createClient` in a `client component`. But I noticed it is sending too many requests in network tabs. below is my code

```"use client";

import { createClient } from "@supabase/supabase-js";

const supabaseBaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseServiceKey = process.env.NEXT_PUBLIC_SUPABASE_SERVICE_KEY!;

export default async function Testing() {
  const supabaseClient = createClient(supabaseBaseUrl, supabaseServiceKey);

  const { data: files, error: filesError } = await supabaseClient
    .from("files")
    .select("id, file_name");
  return (
    <>
     <p>{JSON.stringify(files)}</p>
    </>
  );
}
```

You can also see in the image attached, there are more than **250 requests** in the network tab and it is continuously growing. 
I don't know what's the issue and I couldn't find any solution on Google. Please help me if you know the solution.

I'm also getting the following warning in the conosle

`Multiple GoTrueClient instances detected in the same browser context. It is not an error, but this should be avoided as it may produce undefined behavior when used concurrently under the same storage key.`

<img width="1436" alt="Screenshot 2023-10-23 at 1 04 39 AM" src="https://github.com/supabase/supabase/assets/30028457/efe5440f-5ec9-4cb7-ac74-e43ec9e306bc">

**Comment by GaryAustin1** [comment]
createClient does not normally make requests unless it is restoring an expired user session.  Your requests are from the select.
You seem to be calling this code over and over.  Put a console.log() in there and see if that is the case.
